beacon/blockchain/finalize_block.go

```diff
@@ -98,7 +99,22 @@ func (s *Service) FinalizeBlock(
 
 	// fetch and store the deposit for the block
 	blockNum := blk.GetBody().GetExecutionPayload().GetNumber()
-	s.depositFetcher(ctx, blockNum)
+
+	if isSyncing(req.Height, req.SyncingToHeight) {
+		s.logger.Info(
+			"Syncing in finalize_block, skipping deposit fetcher",
+			"height", req.Height, "syncing_to_height", req.SyncingToHeight,
+		)
+		// If we're syncing, we just use the consensus finalized block deposits as an optimization
+		// The DepositStore does not get included in AppHash calculation and hence introducing this will not result in AppHash.
```

Comment:
note: maybe the long term fix is to include deposits in the state, to be able to roll them back as the rest of the beacon state

Reply:
nit: maybe the comment can be cleaned up? Something like:
`If we're syncing we do not expect the EL state to be up to date and so we don't pull deposits from it. Instead we stored deposits contained in the finalized block`